plugins/pipelines/lib/samson_pipelines/samson_plugin.rb

```diff
@@ -0,0 +1,27 @@
+module SamsonPipelines
+  class Engine < Rails::Engine
+  end
+end
+
+Samson::Hooks.view :stage_form, 'samson_pipelines/fields'
+
+Samson::Hooks.callback :stage_permitted_params do
+  { next_stage_ids: [] }
+end
+
+Samson::Hooks.callback :after_deploy do |deploy|
+  begin
+    return unless deploy.succeeded? || deploy.stage.next_stage_id.empty?
+
+    log = ''
+    Stage.find(deploy.stage.next_stage_ids).each do |next_stage|
+      new_deploy = DeployService.new(deploy.user).deploy!(next_stage, deploy.commit)
+      log += "Kicking off next stage: #{next_stage.name} - URL: #{new_deploy.url}\n"
+    end
+    deploy.job.reload.update_output!(deploy.job.output + "\n#{log}") if log.presence
```

Comment:
You would just append to the log, no db updating necessary

On Tue, Jun 2, 2015 at 12:06 PM, Shane Hender <notifications@github.com>
wrote:

> In plugins/pipelines/lib/samson_pipelines/samson_plugin.rb
> <https://github.com/zendesk/samson/pull/425#discussion_r31559130>:
>
> > +Samson::Hooks.view :stage_form, 'samson_pipelines/fields'
> > +
> > +Samson::Hooks.callback :stage_permitted_params do
> > +  { next_stage_ids: [] }
> > +end
> > +
> > +Samson::Hooks.callback :after_deploy do |deploy|
> > +  begin
> > +    return unless deploy.succeeded? || deploy.stage.next_stage_id.empty?
> > +
> > +    log = ''
> > +    Stage.find(deploy.stage.next_stage_ids).each do |next_stage|
> > +      new_deploy = DeployService.new(deploy.user).deploy!(next_stage, deploy.commit)
> > +      log += "Kicking off next stage: #{next_stage.name} - URL: #{new_deploy.url}\n"
> > +    end
> > +    deploy.job.reload.update_output!(deploy.job.output + "\n#{log}") if log.presence
>
> I didn't want to update the job DB record N times, but instead update the
> job after all pipeline stages are kicked off.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zendesk/samson/pull/425/files#r31559130>.
>